### ISSUE TITLE
Nimbus crashes with "SIGSEGV: Illegal storage access"

### ISSUE THREAD

**Issue by ibhagwan** [status-im/nim-beacon-chain#4123]
**Describe the bug**

Nimbus crashes with:
```
Traceback (most recent call last, using override)
../csu/libc-start.c(314)
/home/user/nimbus-eth2/vendor/nim-libp2p/libp2p/stream/bufferstream.nim(474) main
/home/user/nimbus-eth2/vendor/nim-libp2p/libp2p/stream/bufferstream.nim(467) NimMain
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(2152) main
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(2020) handleStartUpCmd
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1858) doRunBeaconNode
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1660) start
/home/user/nimbus-eth2/beacon_chain/nimbus_beacon_node.nim(1603) run
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncloop.nim(288) poll
/home/user/nimbus-eth2/vendor/nim-chronos/chronos/asyncfutures2.nim(365) futureContinue
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/jsonmarshal.nim(214) engine_exchangeTransitionConfigurationV1
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/jsonmarshal.nim(148) unpackArg
/home/user/nimbus-eth2/vendor/nim-json-rpc/json_rpc/jsonmarshal.nim(40) fromJson
/home/user/nimbus-eth2/vendor/nimbus-build-system/vendor/Nim/lib/system/excpt.nim(610) signalHandler
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

Also experiencing #4122, not sure if issues are related.

**Comment by mikeshultz** [comment]
I'm seeing the same thing.  Restarting gets it a little bit further, but it will happen again within a few minutes.  I've been syncing through the merge, if that matters.

**Comment by arnetheduck** [comment]
If you're experiencing this on a regular basis, it is recommended to switch to websockets instead (using a `ws://` url) - we are testing a fix in the meantime.

**Comment by ibhagwan** [comment]
> If you're experiencing this on a regular basis, it is recommended to switch to websockets instead (using a `ws://` url) - we are testing a fix in the meantime.

Can confirm, using this in my `config.toml` solved the crash issue:
```toml
web3-url = [ "ws://localhost:8551" ]
```

**Comment by mikeshultz** [comment]
This also seems to have worked for me.  Thanks.  Will report back if I see it again.

For anyone else who ends up here with Erigon: the Engine API does not run on the `rpcdaemon` process.  So make sure you configure websockets for the main process as well.

**Comment by zah** [comment]
We believe this issue was fixed in Nimbus 22.9.1 and the currently both HTTP and WebSocket connections to the execution client should be handled without problems. Please, send us a report if you encounter the issue again.
